src/api/clinical/diagnosis/diagnosis.controller.ts

```diff
@@ -115,12 +114,11 @@ export class DiagnosisController extends BaseController{
         try {
 
             const id: uuid = await this._validator.getParamUuid(request, 'id');
-            await this.authorizeOne(request, null, null);
             const existingUser = await this._service.getById(id);
             if (existingUser == null) {
                 throw new ApiError(404, 'Diagnosis record not found.');
             }
-            
+            await this.authorizeOne(request, existingUser.Patient.UserId, existingUser.Patient.TenantId);
```

Comment:
This needs to be looked at. There are some problems.

Reply:
Modified the diagnosis mapper